flink-core/src/main/java/org/apache/flink/configuration/DeploymentOptions.java

```diff
@@ -34,7 +34,13 @@
 			key("execution.target")
 					.stringType()
 					.noDefaultValue()
-					.withDescription("The deployment target for the execution, e.g. \"local\" for local execution.");
+					.withDescription(" The deployment target for the execution. " +
+							"This can take one of the following values:" +
+							"\"remote\", \"local\", \"yarn-per-job\", " +
+							"\"yarn-session\", \"yarn-application\", " +
+							"\"kubernetes-session\" and \"kubernetes-application\". " +
+							"Note that \"yarn-application\" and \"kubernetes-application\" " +
+							"can only be used through the CLI using the \"run-application -t\" command.");
```

Comment:
Should we use the list option of the `Description` here? 

```
Description.builder()
						.text("The deployment target for the execution. This can take one of the following values:")
						.list(text("remote"), text("local"))
						.text("Note: foobar")
						.build()
```